ert_shared/port_handler.py

```diff
@@ -134,7 +138,19 @@ def get_family(host: str) -> socket.AddressFamily:
         return socket.AF_INET
 
 
+# See https://stackoverflow.com/a/28950776
 def _get_ip_address() -> str:
-    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-    s.connect(("8.8.8.8", 80))
-    return s.getsockname()[0]
+    try:
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        s.settimeout(0)
+        # try pinging a reserved, internal address in order
+        # to determine IP representing the default route
+        s.connect(("10.255.255.255", 1))
+        retval = s.getsockname()[0]
+    except BaseException:  # pylint: disable=broad-except
+        logger.warning("Failed getting ip-address. Fallback to localhost...")
```

Comment:
In my mind, this is an error. "Warning: _failed_" But I don't feel strongly about it.

Reply:
Rephrased...